### PR TITLE
[Remove Stream/Optional usage] LPS-175660   headless-common-spi

### PR DESCRIPTION
**Pull request by JanisZhang**
https://issues.liferay.com/browse/LPS-175660

### CODE REVIEW

**Comment by catarineaguiar** [comment]
ci:test:headless

**Comment by catarineaguiar** [comment]
no regressions found.

File: modules/apps/headless/headless-common-spi/src/main/java/com/liferay/headless/common/spi/odata/entity/EntityFieldsUtil.java [hunk 1]
```diff
@@ -54,19 +52,10 @@ public static List<EntityField> getEntityFields(
...
-		List<ExpandoColumn> expandoColumns =
-			expandoColumnLocalService.getColumns(expandoTable.getTableId());
-
-		Stream<ExpandoColumn> expandoColumnsStream = expandoColumns.stream();
-
-		return expandoColumnsStream.map(
+		return TransformUtil.transform(
+			expandoColumnLocalService.getColumns(expandoTable.getTableId()),
 			expandoColumn -> _getEntityField(
-				expandoBridgeIndexer, expandoColumn)
```
Comment (JanisZhang) [inline modules/apps/headless/headless-common-spi/src/main/java/com/liferay/headless/common/spi/odata/entity/EntityFieldsUtil.java:64]: Hi @dantewang, I have simplified the code here. Thanks!

**Comment by JanisZhang** [comment]
Force-pushed.

**Comment by JanisZhang** [comment]
ci:test:sf

**Comment by JanisZhang** [comment]
ci:test:relevant

**Comment by JanisZhang** [comment]
ci:test:headless

**Comment by dantewang** [comment]
ci:forward

cc @shuyangzhou @tinatian 

The newly-pushed change is just SF without logic change, since relevant passed, no need to wait for another QA check.

**Comment by dantewang** [comment]
Closed the forwarded PR and reopened this one, wait for the discussion in Slack.

**Comment by dantewang** [comment]
ci:forward

cc @shuyangzhou @tinatian Re-forward this one according to the Slack discussion. The product team in charge will do the further clean-up in https://issues.liferay.com/browse/LPS-175792.

**Pull request closed without merging**